src/app/components/PageLayout/Header.tsx

```diff
@@ -8,6 +8,23 @@ import { NetworkSelector } from './NetworkSelector'
 import Box from '@mui/material/Box'
 import { useScopeParam } from '../../hooks/useScopeParam'
 import { useScreenSize } from '../../hooks/useScreensize'
+import { fixedLayer, fixedNetwork, scopeFreedom, RouteUtils } from '../../utils/route-utils'
+
+/**
+ * Do we need to display a scope selector, given the current configuration?
+ */
+const needsScopeSelector = (() => {
+  switch (scopeFreedom) {
+    case 'network':
+      return RouteUtils.getEnabledNetworksForLayer(fixedLayer!).length > 1
+    case 'layer':
+      return RouteUtils.getVisibleLayersForNetwork(fixedNetwork!).length > 1
+    case 'both':
+      return RouteUtils.getEnabledScopes().length > 1
+    case 'none':
+      return false
+  }
+})() // We will define and call this function only once, and then store the result.
```

Comment:
Does this needs to be an IIFE? I would move this part to utils, and just call the function here, if it is needed only in Header. Or just get the static result from utils, since it probably won't change.

Reply:
Moved function to route-utils.